docker/nginx/conf.d/nginx-ssl.conf

```diff
@@ -0,0 +1,101 @@
+limit_req_zone $binary_remote_addr zone=std_rl:10m rate=10r/s;
+limit_req zone=std_rl burst=10 nodelay;
+
+upstream gaia_hub {
+  server hub:3000;
+}
+
+upstream gaia_admin {
+  server admin:8009;
+}
+
+upstream gaia_reader {
+  server reader:8008;
+}
+
+
+server {
+  listen ${SERVER_HTTP_PORT};
+  listen [::]:${SERVER_HTTP_PORT};
+  server_name ${DOMAIN};
+  server_tokens off;
+
+  location /.well-known/acme-challenge {
+    root /usr/share/nginx/html/certbot;
+  }
+
+  location / {
+    return 301 https://$host$request_uri;
+  }
+}
+
+server {
+  listen ${SERVER_HTTPS_PORT} ssl;
+  listen [::]:${SERVER_HTTPS_PORT} ssl;
+  server_name ${DOMAIN};
+
+  server_tokens off;
+  ssl_certificate /etc/letsencrypt/live/${DOMAIN}/fullchain.pem;
+  ssl_certificate_key /etc/letsencrypt/live/${DOMAIN}/privkey.pem;
+  include /etc/letsencrypt/options-ssl-nginx.conf;
+  ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem;
+
+  #security headers
+  add_header Strict-Transport-Security "max-age=31536000; includeSubDomains; preload";
+  add_header X-XSS-Protection "1; mode=block" always;
+  add_header X-Content-Type-Options "nosniff" always;
+  add_header X-Frame-Options "DENY" always;
+
+  ##CSP
+  add_header Content-Security-Policy "frame-src 'self'; default-src 'self'; script-src 'self' 'unsafe-inline' https://maxcdn.bootstrapcdn.com https://ajax.googleapis.com; img-src 'self'; style-src 'self' https://maxcdn.bootstrapcdn.com; font-src 'self' data: https://maxcdn.bootstrapcdn.com; form-action 'self'; upgrade-insecure-requests;" always;
+  add_header Referrer-Policy "strict-origin-when-cross-origin" always;
+
```

Comment:
Okay -- I think there's a couple of issues here -- 

The following paths need to go the `gaia_hub` upstream:

`/store/*`, `/hub_info`, `/list-files/*` (and, in the future `/revoke/`)

Alternatively, we can say, send all `/hub/*` traffic to the `gaia_hub` upstream, then delete the `/hub_info` entry from here.

That configuration would give us:

`/hub/` -> points to the gaia hub, which services writes

`/reader/` -> points to the reader. which services reads

Reply:
I"ll work on the nginx config here. i think i understand what you're suggesting and will update later.